pkg/server/application_api/application_api_test: TestStatusAPICombinedStatementsWithFullScans failed

pkg/server/application_api/application_api_test.TestStatusAPICombinedStatementsWithFullScans [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12406027?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12406027?buildTab=artifacts#/) on release-23.2 @ [6aa847e0e6f7eb6ab26dd9b150f165ccae9dd2c6](https://github.com/cockroachdb/cockroach/commits/6aa847e0e6f7eb6ab26dd9b150f165ccae9dd2c6):


```
=== RUN   TestStatusAPICombinedStatementsWithFullScans
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/03e3c3f86914ef5ebcca1a9ffbd26abe/logTestStatusAPICombinedStatementsWithFullScans276154095
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_stats_test.go:653: 
        	Error Trace:	pkg/server/application_api/application_api_test_test/pkg/server/application_api/sql_stats_test.go:653
        	            				pkg/server/application_api/application_api_test_test/pkg/server/application_api/sql_stats_test.go:659
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestStatusAPICombinedStatementsWithFullScans
        	Messages:   	failed for respStatement: {"key":{"key_data":{"query":"CREATE DATABASE football","app":"TestCombinedStatementsWithFullScans","distSQL":false,"failed":false,"implicit_txn":false,"vec":false,"full_scan":false,"database":"defaultdb","plan_hash":0,"query_summary":"CREATE DATABASE football","transaction_fingerprint_id":0},"aggregated_ts":"2021-08-30T19:00:00Z","aggregation_interval":0},"id":8429810229607860816,"stats":{"count":2,"first_attempt_count":1,"max_retries":1,"legacy_last_err":"","legacy_last_err_redacted":"","num_rows":{"mean":0,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"parse_lat":{"mean":0.000109105,"squared_diffs":0},"plan_lat":{"mean":0.0010460325,"squared_diffs":2.4483525612499987e-8},"run_lat":{"mean":0.0744386505,"squared_diffs":0.00003831926068082448},"service_lat":{"mean":0.0756191105,"squared_diffs":0.000040392278724264415},"overhead_lat":{"mean":0.00002532250000000097,"squared_diffs":7.681801250015193e-11},"sensitive_info":{"last_err":"","most_recent_plan_description":{"name":""},"most_recent_plan_timestamp":"0001-01-01T00:00:00Z"},"bytes_read":{"mean":0,"squared_diffs":0},"rows_read":{"mean":0,"squared_diffs":0},"rows_written":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":0,"squared_diffs":0},"max_mem_usage":{"mean":10240,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":0,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":8456120,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":0,"squared_diffs":0},"step_count_internal":{"mean":0,"squared_diffs":0},"seek_count":{"mean":0,"squared_diffs":0},"seek_count_internal":{"mean":0,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":0,"squared_diffs":0},"point_count":{"mean":0,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}}},"sql_type":"","last_exec_timestamp":"2021-08-30T19:50:00Z","nodes":[3],"regions":["test"],"plan_gists":["Ais="],"latency_info":{"min":0.071125099,"max":0.080113122,"p50":0,"p90":0,"p99":0},"last_error_code":""},"txn_fingerprint_ids":[15825384044486114703]}
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/03e3c3f86914ef5ebcca1a9ffbd26abe/logTestStatusAPICombinedStatementsWithFullScans276154095
--- FAIL: TestStatusAPICombinedStatementsWithFullScans (51.53s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/unowned
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatusAPICombinedStatementsWithFullScans.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32843

--- a/pkg/server/application_api/sql_stats_test.go
+++ b/pkg/server/application_api/sql_stats_test.go
@@ -510,6 +510,7 @@ func TestStatusAPICombinedStatementsWithFullScans(t *testing.T) {
 	if skip.Stress() {
 		additionalTimeout = additionalTimeoutUnderStress
 	}
+	skip.UnderStressRace(t, "test is too slow to run under race")
 
 	// Aug 30 2021 19:50:00 GMT+0000
 	aggregatedTs := int64(1630353000)
@@ -637,7 +638,7 @@ func TestStatusAPICombinedStatementsWithFullScans(t *testing.T) {
 			respStatement, exists := actualResponseStatsMap[respQuery]
 			require.True(t, exists, "Expected statement '%s' not found in response: %v", respQuery, responseToJSON(resp))
 
-			actualCount := respStatement.Stats.Count
+			actualCount := respStatement.Stats.FirstAttemptCount
 			actualFullScan := respStatement.Key.KeyData.FullScan
 			actualDistSQL := respStatement.Key.KeyData.DistSQL
 			actualFailed := respStatement.Key.KeyData.Failed

